packages/messaging/src/models/clean-v1-undefined.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * There seems to have been a bug in the messaging SDK versions <= 4.9.x
+ * where the IndexedDB model was using a database name of 'undefined'.
+ *
+ * In 4.10.x we changed the model implementation, but kept the database
+ * name as it should have been. This however introduced an issue where
+ * two tokens were pointing to the same underlying PushSubscription.
+ *
+ * This code will look for the undefined database and delete any of the
+ * underlying tokens.
+ */
+
+import IIDModel from '../models/iid-model';
+
+const OLD_DB_NAME = 'undefined';
+const OLD_OBJECT_STORE_NAME = 'fcm_token_object_Store';
+
+function handleDb(db: IDBDatabase) {
+  if (!db.objectStoreNames.contains(OLD_OBJECT_STORE_NAME)) {
+    // We found a database with the right name, but our expected object
+    // store isn't defined.
+    return;
```

Comment:
this comment is confusing. So the v1 object store doesn't exist, and the new one is not defined yet? does this imply the old object store was dropped (maybe manually), and a new one isn't created yet?

Reply:
This is looking for the database named 'undefined', which *could* be used be created anyone, which if it is - we perhaps may not be the owner and therefore the expected object store wouldnt exist.